libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -49,16 +49,9 @@ Declaration const* DeclarationContainer::conflictingDeclaration(
 		dynamic_cast<MagicVariableDeclaration const*>(&_declaration)
 	)
 	{
-		// check that all other declarations with the same name are functions or a public state variable or events.
-		// And then check that the signatures are different.
+		// check that all other declarations are of the same kind
```

Comment:
Please add a comment that the type checker ensures that the signatures are different.